pyfr/solvers/baseadvec/elements.py

```diff
@@ -99,3 +97,51 @@ def filter_soln(uin):
                 return self._be.ordered_meta_kernel([mul, copy])
 
             kernels['filter_soln'] = filter_soln
+
+        
+        shock_capturing = self.cfg.get('solver', 'shock-capturing', 'none')
+        if shock_capturing == 'entropy-filter':
+            nonce = args[2]
+            tags = {'align'}
+
+            self.entmin     = self._be.matrix((1, self.neles),
+                                           extent=nonce + 'entmin', tags=tags)
+            self.entmin_int = self._be.matrix((self.nfpts, self.neles),
+                                           extent=nonce + 'entminint', tags=tags)
+   
+            eftplargs = {'nfpts' : self.nfpts}
+            self.kernels['min_entropy'] = lambda: self._be.kernel(
+                'entropymin', tplargs=eftplargs, dims=[self.neles],
+                entmin=self.entmin, entmin_int=self.entmin_int
+            )
+
+            # Setup nodal/modal operator matrices
+            self.vdm = self._be.matrix((self.nupts, self.nupts),
+                                           extent=nonce + 'vdm', initval=self.basis.ubasis.vdm.T)
+            self.invvdm = self._be.matrix((self.nupts, self.nupts),
+                                           extent=nonce + 'invvdm', initval=self.basis.ubasis.invvdm.T)
+            
+            # Setup interpolation matrices if applying constraints on fpts/qpts
+            con_fpts = self.cfg.getbool('solver-entropy-filter', 'constrain-fpts', False)
+            con_qpts = self.cfg.getbool('solver-entropy-filter', 'constrain-qpts', False)
+
+            if con_qpts and not self.nqpts:
```

Comment:
I'd neglect this check as without flux AA there are no qpts, right? Just makes it easier to toggle anti-aliasing on and off without needing to change another parameter.